commitpkg.in

```diff
@@ -83,18 +83,23 @@ while getopts ':l:a:s:f' flag; do
 done
 shift $(( OPTIND - 1 ))
 
-# check packages have the packager field set
+# check packages for validity
 for _arch in "${arch[@]}"; do
 	if [[ -n $commit_arch && ${_arch} != "$commit_arch" ]]; then
 		continue
 	fi
 	for _pkgname in "${pkgname[@]}"; do
 		fullver=$(get_full_version "$_pkgname")
 
-		if pkgfile=$(find_cached_package "$_pkgname" "$_arch" "$fullver"); then
+		if pkgfile=$(find_cached_package "$_pkgname" "$fullver" "$_arch"); then
 			if grep -q "packager = Unknown Packager" <(bsdtar -xOqf "$pkgfile" .PKGINFO); then
 				die "PACKAGER was not set when building package"
 			fi
+			hashsum=sha256sum
+			pkgbuild_hash=$(grep "^pkgbuild_$hashsum" <(bsdtar -xOqf "$pkgfile" .BUILDINFO) | sed -E 's|pkgbuild_.+ = (.+)|\1|')
```

Comment:
```suggestion
			pkgbuild_hash=$(awk -v"hashsum=$hashsum" -F' = ' '$1 == "pkgbuild_"hashsum {print $2}' <(bsdtar -xOqf "$pkgfile" .BUILDINFO))
```

awk understands field records already. <3

If field 1 is pkgbuild_$hashsum, then print field 2. Much easier to understand than sed. Also it does not use regex, anywhere, just full string equivalence tests.